tsup.config.ts

```diff
@@ -0,0 +1,9 @@
+import { defineConfig } from 'tsup';
```

Comment:
esbuild for esbuild is a little funny, but do we need to use this? I'd rather stick to what the rest of the repositories in our organization use (plain old tsc) if it'll work the same. I'm cautious about introducing new dependencies that we might not need or mismatching our build systems in the ecosystem (everywhere else uses plain ol `tsc`).

Reply:
uh hm

Reply:
update: i think we can use tsc but ill check